idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/inspections/explicitcoercion.kt

```diff
@@ -0,0 +1,102 @@
+package arrow.meta.ide.plugins.proofs.inspections
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.dsl.IdeSyntax
+import arrow.meta.ide.plugins.proofs.markers.participatingTypes
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import arrow.meta.plugins.proofs.phases.coerceProof
+import arrow.meta.quotes.ktFile
+import com.intellij.codeHighlighting.HighlightDisplayLevel
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.idea.imports.importableFqName
+import org.jetbrains.kotlin.idea.inspections.AbstractApplicabilityBasedInspection
+import org.jetbrains.kotlin.psi.KtElement
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtImportDirective
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.resolve.ImportPath
+import org.jetbrains.kotlin.types.KotlinType
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+/**
+ * [explicitCoercionInspection]: for implicit coercion to make it explicit
+ */
+val IdeMetaPlugin.explicitCoercionInspection: ExtensionPhase
+  get() = addLocalInspection(
+    inspection = explicitCoercionInspectionSyntax,
+    level = HighlightDisplayLevel.WEAK_WARNING,
+    groupPath = ProofPath + arrayOf("Coercion")
+  )
+
+const val EXPLICIT_COERCION_INSPECTION_ID = "Make_coercion_explicit"
+
+val IdeSyntax.explicitCoercionInspectionSyntax: AbstractApplicabilityBasedInspection<KtElement>
+  get() = applicableInspection(
+    defaultFixText = EXPLICIT_COERCION_INSPECTION_ID,
+    inspectionHighlightType = { ProblemHighlightType.INFORMATION },
+    kClass = KtElement::class.java,
+    inspectionText = { "Not used at the moment because the highlight type used is ProblemHighlightType.INFORMATION" },
+    applyTo = { ktCall: KtElement, _, _ ->
```

Comment:
@i-walker is there a reason for the current param ordering in this method? could we reorder to have `isApplicable` before `applyTo` on the DSL declaration?

Reply:
these are just named parameters (so we can sort them as we want).. although having certain order can help to reason about it 👍

Reply:
the order doesn't matter from a lib perspective

Reply:
Yes Dani! I was taking into account that named params can be ordered at will on call sites, the question is mostly so we could have them "naturally" ordered for us on the declaration too. If you autocomplete they are always inferred / proposed in the order they are declared. Imho that's part of the library public api design and would help both to library contributors and us.

Reply:
those are autocompleted automatically. But when we started this DSL I left those parameters named for a better learning curve for everyone learning and building ide-plugins. We can remove those in the implementation parts of Meta

Reply:
nooo, lets keep the names.. it helps!

Reply:
It makes reviewing a lot simpler. Without remembering what parameter is in which position